src/Engine/ProtoCore/Parser/AssociativeAST.cs

```diff
@@ -1287,6 +1289,26 @@ public MethodAttributes(bool hiddenInLibrary = false, string msg = "")
         }
     }
 
+    public class ExternalAttributes
```

Comment:
We have `MethodAttributes`, `ClassAttributes` classes, and each one expose some properties that *only related to FFI*, e.g., `HiddenInLibrary`. 

As attributes essentially is just a list of key value pair, it'd better to unify them into a single class, here is `ExternalAttributes`, which basically is a simple wrapper for a dictionary. Then `FFIClassAttributes`, `FFIMethodAttributes` and `FFIParameterAttributes` can derive from it, and for these FFI attribute classes, they are fine to have properties. E.g.,
```
public class FFIParamAttributes: ProtoCore.AST.AssociativeAST.ExternalAttributes
{
    public bool IsArbitraryDimensionArray { get { ... }}
    public string DefaultArgumentExpression { get { ... } }
}
```